repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala

```diff
@@ -102,7 +105,30 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
         wrappedRequest.setRequestURI(joinTokensIntoPath(uriTokens))
       }
 
-      filterChain.doFilter(wrappedRequest, wrappedResponse)
+      val requestLinkPaths = config.getLinkResource
+        .filter(resource => resource.getUriPathRegex =~ wrappedRequest.getRequestURI)
+        .filter(resource => isMatchingMethod(resource.getHttpMethods.toSet, wrappedRequest.getMethod))
+        .map(resource => getPathsForContentType(wrappedRequest.getContentType, resource.getRequest))
+        .fold(List.empty[LinkPath])(_ ++ _)
+
+      val filterChainRequest = {
+        if (token.isDefined && requestLinkPaths.nonEmpty) {
+          Option(wrappedRequest.getContentType) match {
+            case Some(ct) if ct.toLowerCase.contains("json") => handleJsonRequestLinks(wrappedRequest, previousToken, nextToken, token, requestLinkPaths)
```

Comment:
For this, they aren't used... yet. This was the potential to use them contextually like on the response in the future. It also let us easily reuse the response methods with only a minor tweak in the logic. I didn't want to throw away a bunch of references that we didn't NEED right now when there was a big potential they would be needed later. So since they were there and the reuse was easier with them, I kept them.